### PR TITLE
[User Attribution] "createdBy" and "modifiedBy" fields for domainObjects

### LINKED ISSUE #5656: Add user author attribution to objects
**Is your feature request related to a problem? Please describe.**
Current objects have a modified and a persisted time associated with them. We should also a user attribution to objects, as we now do with notebook entries.

**Describe the solution you'd like**
"createdBy" and "modifiedBy" fields could be added and automatically populated with the logged in user, if available.

### LINKED ISSUE #5801: [objectApi] Add "created" timestamp to newly created domainObjects
**Is your feature request related to a problem? Please describe.**
No we have created timestamp in: https://github.com/nasa/openmct/blob/master/src/plugins/persistence/couch/CouchObjectProvider.js#L657 but it's immediately lost upon save. Not a problem per se, but makes this bit of code pointless. 

**Describe the solution you'd like**
We add a created attribute to objects when we save them for the first time in the objectApi. Remove the non working code above. This information will need to be added/shown to the inspector as well when viewing information related to a domainObject.

**Describe alternatives you've considered**
Currently, as mentioned above, we try to save it to the CouchObjectProvider metadata, but it's lost immediately.

**Additional context**
We have persisted and modified timestamps currently on domainObjects and we'll be adding createdBy and modifiedBy attribution as well, this seems like something that we should also have.

### LINKED ISSUE #5818: [Restricted Notebook] Sync changes
**Is your feature request related to a problem? Please describe.**
We currently sync changes from couch for a couple of object types, _notably_ Notebooks.

**Describe the solution you'd like**
We need to add the object type for restricted notebooks as well.

**Additional context**
This was intended initially, we just missed this one.

## Testing
https://github.com/nasa/openmct/issues/3784#issuecomment-848086390
Same as above, except you will also want to lock notebook pages as well and verify that also syncs.

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #5656 
Closes #5801 
Closes #5818
### Describe your changes:

Grabbing current user when saving or updating objects. Whenever a new object is saved, we check if it's new and add a `createdBy` field and if it is being updated, we add/update the `modifiedBy` field. Also, adding a `created` timestamp to each newly created object.

Also, since working in the same file, closing out issue #5818, adding restricted notebooks to our currently synced objects (note all objects are to be synced in the future).

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/api/objects/ObjectAPI.js [hunk 1]
```diff
@@ -365,6 +373,7 @@ export default class ObjectAPI {
                     savedReject = reject;
                 });
                 domainObject.persisted = persistedTime;
+                domainObject.createdBy = this.currentUsername;
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:376]: There's the potential for a race condition here. It's conceivable that the user would not have been resolved at this point if the user provider is slower than the object provider. Object save can happen automatically on load when migration scripts update objects from legacy models. I'd also prefer not to cache the logged in user anywhere but the user provider, and leave control of the caching strategy up to the user provider itself.

So, TL;DR, let's call the User API here and fetch the user from there each time. `save` is asynchronous so it's ok to do it here.

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/api/objects/ObjectAPI.js (hunk 1 above)
Comment (jvigliotta) [reply src/api/objects/ObjectAPI.js]: sigh... duh. Thank you!

File: src/api/objects/ObjectAPI.js [hunk 2]
```diff
@@ -90,6 +90,14 @@ export default class ObjectAPI {
 
         this.SYNCHRONIZED_OBJECT_TYPES = ['notebook', 'plan', 'annotation'];
 
+        this.currentUsername = undefined;
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:93]: don't believe we need this any more

File: src/api/objects/ObjectAPI.js [hunk 3]
```diff
@@ -323,6 +331,7 @@ export default class ObjectAPI {
     }
 
     isPersistable(idOrKeyString) {
+        console.log('isPersistable');
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:334]: debug message

File: src/api/objects/ObjectAPI.js [hunk 4]
```diff
@@ -347,7 +356,8 @@ export default class ObjectAPI {
      * @returns {Promise} a promise which will resolve when the domain object
      *          has been saved, or be rejected if it cannot be saved
      */
-    save(domainObject) {
+    async save(domainObject) {
+        console.log('save');
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:360]: debug message

File: src/api/objects/ObjectAPI.js [hunk 5]
```diff
@@ -481,6 +505,7 @@ export default class ObjectAPI {
      * @memberof module:openmct.ObjectAPI#
      */
     mutate(domainObject, path, value) {
+        console.log('mutate');
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:508]: debug message

File: src/api/objects/ObjectAPI.js [hunk 6]
```diff
@@ -392,6 +405,17 @@ export default class ObjectAPI {
         });
     }
 
+    async #getCurrentUsername() {
+        const user = await this.openmct.user.getCurrentUser();
+        let username = 'Unknown';
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:410]: Let's just leave this undefined if the user is unknown. All of our legacy objects will have undefined users, and we have logic for handling that case in the Vue component.

**Review by akhenry — CHANGES_REQUESTED** [review]
Looking really good, thanks for making these changes. Couple of suggestions.

**Review by akhenry — APPROVED** [review]
